cvat/apps/engine/rq_job_handler.py

```diff
@@ -28,82 +32,85 @@ class RQJobMetaField:
 def is_rq_job_owner(rq_job: RQJob, user_id: int) -> bool:
     return rq_job.meta.get(RQJobMetaField.USER, {}).get('id') == user_id
 
-@attrs.define(kw_only=True)
+@attrs.frozen()
 class RQId:
-    action: str = attrs.field(
-        validator=attrs.validators.instance_of(str)
+    action: RequestAction = attrs.field(
+        validator=attrs.validators.instance_of(RequestAction)
     )
-    resource: str = attrs.field(
-        validator=attrs.validators.instance_of(str)
+    target: RequestTarget = attrs.field(
+        validator=attrs.validators.instance_of(RequestTarget)
     )
     identifier: Union[int, UUID] = attrs.field(
         validator=attrs.validators.instance_of((int, UUID))
     )
-    subresource: Optional[str] = attrs.field(
+    subresource: Optional[RequestSubresource] = attrs.field(
         validator=attrs.validators.optional(
-            attrs.validators.instance_of(str)
-        )
+            attrs.validators.instance_of(RequestSubresource)
+        ),
+        kw_only=True, default=None,
     )
     user_id: Optional[int] = attrs.field(
-        validator=attrs.validators.optional(attrs.validators.instance_of(int))
+        validator=attrs.validators.optional(attrs.validators.instance_of(int)),
+        kw_only=True, default=None,
     )
     format: Optional[str] = attrs.field(
-        validator=attrs.validators.optional(attrs.validators.instance_of(str))
+        validator=attrs.validators.optional(attrs.validators.instance_of(str)),
+        kw_only=True, default=None,
     )
 
+    def __attrs_post_init__(self) -> None:
+        if self.action == RequestAction.CREATE and self.target != RequestTarget.TASK:
```

Comment:
there are also a few more cases to be handled:
```suggestion
       if (
            self.action == RequestAction.CREATE
            and self.target != RequestTarget.TASK
            or self.action == RequestAction.IMPORT
            and (
                self.target != RequestTarget.PROJECT
                and self.subresource == RequestSubresource.DATASET
                or self.target == RequestTarget.PROJECT
                and self.subresource == RequestSubresource.ANNOTATIONS
            )
            or self.action in (RequestAction.EXPORT, RequestAction.IMPORT)
            and self.target == RequestTarget.JOB
            and self.subresource == RequestSubresource.BACKUP
        ):
```

Reply:
I decided to redo this function entirely. My original check was adapted from an assert in `RQIdManager.build`, but on second thought, I don't think it's really needed, because it's not `RQId` responsibility to decide which actions can be applied to which resources.

OTOH, it _is_ its responsibility to make sure that the ID can be rendered to a string that can then be parsed back to the same ID. So I added checks that make sure certain fields are present/absent, depending on the action.